tests/suite/tracing_test.go

```diff
@@ -0,0 +1,270 @@
+			worldURL = fmt.Sprintf("%s:%s/world", url, strconv.Itoa(portFwdPort))
+			helloworldURL = fmt.Sprintf("%s:%s/helloworld", url, strconv.Itoa(portFwdPort))
+		}
+	})
+
+	AfterEach(func() {
+		output, err := uninstallCollector()
+		Expect(err).ToNot(HaveOccurred(), string(output))
+
+		Expect(resourceManager.DeleteFromFiles(files, ns.Name)).To(Succeed())
+		Expect(resourceManager.DeleteNamespace(ns.Name)).To(Succeed())
+	})
+
+	updateGatewayClass := func() error {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.CreateTimeout)
+		defer cancel()
+
+		key := types.NamespacedName{Name: "nginx"}
+		var gwClass gatewayv1.GatewayClass
+		if err := k8sClient.Get(ctx, key, &gwClass); err != nil {
+			return err
+		}
+
+		gwClass.Spec.ParametersRef = &gatewayv1.ParametersReference{
+			Group: ngfAPI.GroupName,
+			Kind:  gatewayv1.Kind("NginxProxy"),
+			Name:  "nginx-proxy",
+		}
+
+		return k8sClient.Update(ctx, &gwClass)
+	}
+
+	sendTraceRequests := func(ctx context.Context, url string, count int) {
+		for range count {
+			status, _, err := framework.GetWithRetry(ctx, url, address, timeoutConfig.RequestTimeout)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(status).To(Equal(http.StatusOK))
+		}
+	}
+
+	It("sends tracing spans for one policy attached to one route", func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		sendTraceRequests(ctx, helloURL, 5)
+
+		// verify that no traces exist yet
+		logs, err := resourceManager.GetPodLogs(collectorNamespace, collectorPodName, &core.PodLogOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(logs).ToNot(ContainSubstring("service.name: Str(ngf:helloworld:gateway:my-test-svc)"))
+
+		// install tracing configuration
+		traceFiles := []string{
+			"tracing/nginxproxy.yaml",
+			"tracing/policy-single.yaml",
+		}
+		Expect(resourceManager.ApplyFromFiles(traceFiles, ns.Name)).To(Succeed())
+		Expect(updateGatewayClass()).To(Succeed())
+
+		Eventually(
+			func() error {
+				return verifyGatewayClassResolvedRefs()
+			}).
+			WithTimeout(timeoutConfig.GetTimeout).
+			WithPolling(500 * time.Millisecond).
+			Should(Succeed())
+
+		Eventually(
+			func() error {
+				return verifyPolicyStatus()
+			}).
+			WithTimeout(timeoutConfig.GetTimeout).
+			WithPolling(500 * time.Millisecond).
+			Should(Succeed())
+
+		// send traffic and verify that traces exist for hello app
+		findTraces := func() bool {
+			sendTraceRequests(ctx, helloURL, 25)
+			sendTraceRequests(ctx, worldURL, 25)
+			sendTraceRequests(ctx, helloworldURL, 25)
+
+			logs, err := resourceManager.GetPodLogs(collectorNamespace, collectorPodName, &core.PodLogOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return strings.Contains(logs, "service.name: Str(ngf:helloworld:gateway:my-test-svc)")
+		}
+
+		// wait for expected first line to show up
+		Eventually(findTraces, "1m", "5s").Should(BeTrue())
```

Comment:
would it make sense to split send and find traces? so that we send only once but find eventually (possible multiple times)

Reply:
I had that originally, but sometimes it takes multiple requests for anything to show up in the logs, so I needed to keep resending.